src/http.ts

```diff
@@ -124,11 +125,11 @@ export default class HTTP {
   /**
    * @private
    */
-  async processResponse(response: Response): Promise<HttpResponse> {
+  async processResponse<T>(response: Response): Promise<HttpResponse<T>> {
     const { status, headers } = response;
     const text = await response.text();
     // Check if we have a body; if so parse it as JSON.
-    let json;
+    let json: unknown;
```

Comment:
I guess it should be `T` here

Reply:
Well at this point in the code, it's not known whether `json` is `T` or `ServerResponseObject`. By leaving it `unknown`, we can prevent any unsafe access until we determine which type it might be. If we switch it to `T`, we'll need to cast it back to `unknown` before casting it to `ServerResponseObject`. I think it makes a little more sense to leave it unknown, but I'll defer to your judgement there!